File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        // int minPrice = INT_MAX;
        // int maxProfit = 0;

        // for(int price: prices){
        //     if(price < minPrice){
        //         minPrice = price;
        //     }else if(price - minPrice >= maxProfit){
        //         maxProfit = price - minPrice;
        //     }
        // }

        int minPrice = prices[0], profit = 0, maxProfit = 0;
        for(int i = 1; i <= prices.size() - 1; i++){
            profit = prices[i] - minPrice;
            maxProfit = max(profit, maxProfit);
            minPrice = min(prices[i], minPrice);
        }
        return maxProfit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let minPrice = prices[0];
    let profit = 0;
    for(let i = 1; i < prices.length ; i++){
        profit = Math.max(profit, (prices[i] - minPrice));
        minPrice = Math.min(prices[i] , minPrice);
    }
    return profit;
};